File: docs/components_page/components/progress/animated.jl
using DashBootstrapComponents, DashHtmlComponents

progress = html_div([
    dbc_progress(value = 80, id = "animated-progress", animated = false, striped = true),
    dbc_button(
        "Toggle animation",
        id = "animation-toggle",
        className = "mt-3",
        n_clicks = 0,
    ),
]);

callback!(
    app,
    Output("animated-progress", "animated"),
    Input("animation-toggle", "n_clicks"),
    State("animated-progress", "animated"),
) do n, animated
    return n > 0 ? animated == 0 : animated
end;

File: docs/components_page/components/progress/animated.R
library(dashBootstrapComponents)

progress <- div(
  list(
    dbcProgress(
      value = 80, id = "animated-progress", striped = TRUE, animated = FALSE
    ),
    dbcButton(
      "Toggle animation",
      id = "animation-toggle", n_clicks = 0,
      className = "mt-3"
    )
  )
)


app$callback(
  output("animated-progress", "animated"),
  list(
    input("animation-toggle", "n_clicks"),
    state("animated-progress", "animated")
  ),
  function(n, animated) {
    if (n > 0) {
      return(!animated)
    }
    return(animated)
  }
)
